PythonVisualizations/Stack.py

```diff
@@ -0,0 +1,322 @@
+        else:
+            rectPos = indexOrCoords
+            valPos = divide_vector(add_vector(rectPos[:2], rectPos[2:]), 2)
+
+        if color is None:
+            # Take the next color from the palette
+            color = drawable.palette[Stack.nextColor]
+            Stack.nextColor = (Stack.nextColor + 1) % len(drawable.palette)
+        cell_rect = self.canvas.create_rectangle(
+            *rectPos, fill=color, outline='', width=0)
+        cell_val = self.canvas.create_text(
+            *valPos, text=str(key), font=self.VALUE_FONT, fill=self.VALUE_COLOR)
+        handler = lambda e: self.setArgument(str(key))
+        for item in (cell_rect, cell_val):
+            self.canvas.tag_bind(item, '<Button>', handler)
+
+        return cell_rect, cell_val
+
+    def display(self):
+        self.canvas.delete("all")
+
+        for i in range(self.size):  # Draw grid of cells
+            self.createArrayCell(i)
+
+        # go through each Drawable in the list
+        for i, n in enumerate(self.list):
+            print("this is i and n", i, n)
+            # create display objects for the associated Drawables
+            n.display_shape, n.display_val = self.createCellValue(
+                i, n.val, n.color)
+            n.color = self.canvas.itemconfigure(n.display_shape, 'fill')
+
+        self.window.update()
+
+    def makeButtons(self):
+        vcmd = (self.window.register(numericValidate),
+                '%d', '%i', '%P', '%s', '%S', '%v', '%V', '%W')
+
+        # numArguments decides the side where the button appears in the operations grid
+        pushButton = self.addOperation(
+            "Push", lambda: self.clickPush(), numArguments=1)  # validationCmd=vcmd)
+        popButton = self.addOperation(
+            "Pop", lambda: self.clickPop())
+        peekButton = self.addOperation(
+            "Peek", lambda: self.clickPeek())
+        newStackButton = self.addOperation(
+            "New", lambda: self.clickNewStack(), numArguments=1, validationCmd=vcmd)
+
+        return [pushButton, popButton, peekButton, newStackButton]
+
+    def validArgument(self):
+        entered_text = self.getArgument()
+        if entered_text:
+            if len(entered_text) <= self.maxArgWidth:
+                return entered_text
+            else:
+                self.setMessage("Error! {} value is too long".format(entered_text))
+
+    # Button functions
+    def clickPush(self):
+        val = self.validArgument()
+        if val is None:
+            # self.setMessage("Input value must be an integer from 0 to 99.")
+            self.setMessage("Input value must be of a string of length less than", self.maxArgWidth, "characters.")
+        else:
+            if len(self.list) >= self.size:
+                self.setMessage("Error! Stack is already full.")
+            else:
+                self.setButtonsStatus(DISABLED)
+                self.push(val)
+                self.setButtonsStatus(NORMAL)
```

Comment:
Wrapping the call to `self.push(val)` in calls to disable and enable the buttons seems redundant with the actions of the `startAnimations()` and `stopAnimations()` calls inside of `push()` (the call to `cleanUp()` at the end calls `stopAnimations()`).  The start and stop calls disable and enable all buttons except Pause/Play and Stop.  So, I don't thinks the calls to `setButtonsStatus()` are needed here or in the other `click` methods.